cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -1227,6 +1231,51 @@ private boolean hasRebalanceAnnotation(KafkaRebalance kafkaRebalance) {
                 kafkaRebalance.getMetadata().getAnnotations().containsKey(ANNO_STRIMZI_IO_REBALANCE);
     }
 
+    private boolean shouldRenewRebalance(KafkaRebalance kafkaRebalance, KafkaRebalanceAnnotation rebalanceAnnotation) {
+        return  rebalanceAnnotation == KafkaRebalanceAnnotation.refresh || CruiseControlIssues.checkForMatch(kafkaRebalance.getStatus().getConditions());
+    }
+
+    /**
+     * Contains the issues that will be picked up in the periodic reconciliation, once corrected.
+     */
+    private enum CruiseControlIssues {
+
+        /**
+         * The Kafka cluster label is missing .
+         */
+        clusterLabelMissing("Cluster Label missing"),
+
+        /**
+         * Cruise Control was not declared in the Kafka Resource
+         */
+        cruiseControlDisabled("Cruise Control disabled"),
+
+        /**
+         * Requests made to Cruise COntrol while it was starting
+         */
+        connectException("ConnectException");
```

Comment:
is this not used anywhere?

Reply:
This exception occurs only when we apply the KR resource and CC pod is started but not in ready state.

Reply:
But yeah Iam not generating this warning/exception myself

Reply:
Yeah but I don't see this enum value to be used somewhere, is it really needed? If yes, let's use it.

Reply:
Yeah I guess this enum would be needed since when we enable the CC then the pods wouldn't become ready instantaneously and the status would contain `ConnectException` in that case and resource will become `NotReady`. So if we check that if the kafka rebalance status contains `ConnectException` then it need to handled in periodic reconciliation

Reply:
I did tried that, But I was not able to know where this `ConnectException` is being added to the status. TBH i I think this is an exception which is being added in status as a warning.

Reply:
The reason I made this enum value was because when we run the this method:
        public static boolean checkForMatch(List<Condition> conditions) {
            for (CruiseControlIssues issue : CruiseControlIssues.values()) {
                if (conditions.stream().anyMatch(condition -> issue.getReason().equals(condition.getReason()))) {
                    return true;
                }
            }
            return false;
        }

We need this value to see if our rebalance status contains this as reason

Reply:
We need this value to see if our rebalance status contains this as reason

Reply:
`CruiseControlIssues.values()` should contain the `ConnectException` enum value. so when we can compare it with the condition present in the rebalance status

Reply:
But for the `ConnectException` to be in the rebalance status, it means someone puts it there by catching any connection exception to Cruise Control when it's not ready. I don't see any code building a rebalance status condition with such a condition, or? what am I missing?

Reply:
Maybe this method is giving the Connect Exception

Reply:
```    private void httpExceptionHandler(Promise<? extends CruiseControlResponse> result, Throwable t) {
        if (t instanceof TimeoutException) {
            // Vert.x throws a NoStackTraceTimeoutException (inherits from TimeoutException) when the request times out
            // so we catch and raise a TimeoutException instead
            result.fail(new TimeoutException(t.getMessage()));
        } else if (t instanceof ConnectException) {
            // Vert.x throws a AnnotatedConnectException (inherits from ConnectException) when the request times out
            // so we catch and raise a ConnectException instead
            result.fail(new ConnectException(t.getMessage()));
        } else {
            result.fail(t);
        ```

Reply:
`_But for the ConnectException to be in the rebalance status, it means someone puts it there by catching any connection exception to Cruise Control when it's not ready. I don't see any code building a rebalance status condition with such a condition, or? what am I missing?_` Yeah that's exactly I was trying to figure out but was unable to find out. I was just able to see the warning in the status

Reply:
What was the outcome of this conversation?

Reply:
From what I can see this Exception is in the enum so that the KRCR will be continually be put into `New` state every time it is seen, so that it will keep trying to reconcile until CC appears?

Reply:
Yes. I guess it would be required because when we bring cruise control up it will take time to restart and if requests are made during that point of time then we will move to 'NotReady' state due to the `ConnectException`